File: LeetCode/Java/src/main/java/org/redquark/tutorials/leetcode/ReverseInteger.java
package org.redquark.tutorials.leetcode;

/**
 * @author Anirudh Sharma
 */
public class ReverseInteger {

    private static int reverse(int x) {
        boolean isNegative = false;
        if (x < 0) {
            isNegative = true;
            x = -x;
        }
        long reverse = 0;
        while (x > 0) {
            reverse = reverse * 10 + x % 10;
            x /= 10;
        }
        if (reverse > Integer.MAX_VALUE) {
            return 0;
        }
        return (int) (isNegative ? -reverse : reverse);
    }

    public static void main(String[] args) {
        System.out.println(reverse(321));
        System.out.println(reverse(-45315));
        System.out.println(reverse(-450));
        System.out.println(reverse(Integer.MAX_VALUE));
    }
}

File: LeetCode/Kotlin/src/main/kotlin/org/redquark/tutorials/leetcode/ReverseInteger.kt
package org.redquark.tutorials.leetcode

fun reverse(x: Int): Int {
    var num = x
    var isNegative = false
    if (num < 0) {
        isNegative = true
        num = -num
    }
    var reverse: Long = 0
    while (num > 0) {
        reverse = reverse * 10 + num % 10
        num /= 10
    }
    return if (reverse > Int.MAX_VALUE) {
        0
    } else (if (isNegative) -reverse else reverse).toInt()
}

fun main() {
    println(reverse(321))
    println(reverse(-45315))
    println(reverse(-450))
    println(reverse(Int.MAX_VALUE))
}